### PR TITLE
Clean up and correct properties to producer and consumers created by Functions/Sinks/Sources

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

The metadata specified in producers and consumers created by functions, sinks, and sources can help with debugging.  

This PR aims improving that aspect as well as add the metadata to python functions as well

### Modifications
Example of what the meta looks like:

```
        "metadata" : {
          "instance_id" : "0",
          "application" : "pulsar-function",
          "id" : "public/default/test2"
        }
```

### CODE REVIEW

**Comment by jerrypeng** [comment]
@cckellogg please take a look as well

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/InstanceUtils.java [hunk 1]
```diff
@@ -103,4 +106,21 @@
...
+                properties.put("application", "pulsar-function");
+                break;
+            case SOURCE:
+                properties.put("application", "pulsar-source");
+                break;
+            case SINK:
+                properties.put("application", "pulsar-sink");
+                break;
+        }
+        properties.put("id", fullyQualifiedInstanceId);
```
Comment (cckellogg) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/InstanceUtils.java]: should we have a name field too?

Comment (jerrypeng) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/InstanceUtils.java]: we could have separate fields for tenant, namespace, and name.  What do you think?

Comment (cckellogg) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/InstanceUtils.java]: I think just a field with fully qualified name is fine
maybe:
application = pulsar-function|pulsar-sink|pulsar-source
id = tenant/namespace/name
instanceId = instanceId

what do you think?

Comment (jerrypeng) [reply pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/InstanceUtils.java]: @cckellogg sounds good. I have updated the PR

**Pull request merged**